### PR TITLE
Some breakpoint fixes

### LINKED ISSUE #219: Cannot add breakpoint at function
### Setup:
``` jl
julia> using Combinatorics, Debugger

julia> @enter collect(combinations(1:3,2))
In collect(itr) at array.jl:544
>544  collect(itr) = _collect(1:1 #= Array =#, itr, IteratorEltype(itr), IteratorSize(itr))

About to run: (Colon())(1, 1)
1|julia> itr
Combinatorics.Combinations{UnitRange{Int64}}(1:3, 2)
```

### Attempts at adding breakpoints based on documentation
All of these `bp add` attempts crash the debugger.

#### By types of local variable
* `bp add func(x, y)`: add a breakpoint to the method matching the types of the local variable x, y etc
``` jl
1|debug> bp add Base.iterate(itr)
ERROR: MethodError: no method matching get_function_in_module_or_Main(::Module, ::Expr)
```

#### By signature
* `bp add func(::Float64, Int)`: add a breakpoint to methods matching the signature
``` jl
1|debug> bp add Base.iterate(::Combinations)                                                             
ERROR: UndefVarError: Combinations not defined
```
``` jl
1|debug> bp add Base.iterate(::Combinatorics.Combinations)                                               
ERROR: UndefVarError: Combinatorics not defined
```
``` jl
1|debug> bp add Base.iterate(::Combinatorics.Combinations{UnitRange{Int64}})                             
ERROR: UndefVarError: Combinatorics not defined
```

Also tried importing Combinatorics in the debugger instance, which fixes the "Combinatorics not defined" error, but then crashes with another error.
``` jl
1|julia> using Combinatorics

1|debug> bp add Base.iterate(::Combinatorics.Combinations)
ERROR: MethodError: no method matching get_function_in_module_or_Main(::Module, ::Expr)
```

### PR DESCRIPTION
**Pull request by KristofferC**
```
julia> @enter collect(combinations(1:3,2))
In collect(itr) at array.jl:558
>558  collect(itr) = _collect(1:1 #= Array =#, itr, IteratorEltype(itr), IteratorSize(itr))

About to run: (Colon())(1, 1)
1|debug> bp add Base.iterate(itr)
[ Info: added breakpoint for method iterate(::Combinatorics.Combinations{UnitRange{Int64}})
1] iterate(::Combinatorics.Combinations{UnitRange{Int64}})

1|debug> bp add Base.iterate(::Combinatorics.Combinations)
[ Info: added breakpoint for method iterate(::Combinatorics.Combinations)
1] iterate(::Combinatorics.Combinations{UnitRange{Int64}})
2] iterate(::Combinatorics.Combinations)

1|debug> bp add Base.iterate(::Combinatorics.Combinations{UnitRange{Int64}})
[ Info: added breakpoint for method iterate(::Combinatorics.Combinations{UnitRange{Int64}})
1] iterate(::Combinatorics.Combinations{UnitRange{Int64}})
2] iterate(::Combinatorics.Combinations)

1|julia> typeof(itr)
Combinatorics.Combinations{UnitRange{Int64}}
```

Fixes #219

### CODE REVIEW

File: src/breakpoints.jl [hunk 1]
```diff
@@ -66,16 +66,21 @@ function add_breakpoint!(state::DebuggerState, cmd::AbstractString)
...
-        f = nothing
+        has_args = false
         if location_expr isa Symbol
             fsym = location_expr
             f = get_function_in_module_or_Main(m, fsym)
         else
             # check that the expr is a chain of getproperty calls
             expr = fsym = location_expr
-            while expr isa Expr
+            if expr isa Expr && expr.head == :call
```
Comment (timholy) [inline src/breakpoints.jl]: `isexpr(expr, :call)`?

Comment (KristofferC) [reply src/breakpoints.jl]: :+1:

**Comment by milesfrain** [comment]
Crashing when attempting to set the last breakpoint.
Traced this issue down to using `Combinatorics #master`.

``` jl
julia> using Combinatorics, Debugger

julia> @enter collect(combinations(1:3,2))
In collect(itr) at array.jl:601
 600  function collect(itr::Generator)
>601      isz = IteratorSize(itr.iter)
 602      et = @default_eltype(itr)
 603      if isa(isz, SizeUnknown)
 604          return grow_to!(Vector{et}(), itr)
 605      else

About to run: (getproperty)(<suppressed 212 bytes of output>, :iter)
1|debug> bp add Base.iterate(itr)
[ Info: added breakpoint for method iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})
1] iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})

1|debug> bp add Base.iterate(::Combinatorics.Combinations)
[ Info: added breakpoint for method iterate(::Combinatorics.Combinations)
1] iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})
2] iterate(::Combinatorics.Combinations)

1|debug> bp add Base.iterate(::Combinatorics.Combinations{UnitRange{Int64}})
ERROR: TypeError: in Type{...} expression, expected UnionAll, got Type{Combinatorics.Combinations}
Stacktrace:
 [1] collect(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}}) at array.jl:601
```

Also not halting when continuing. This is for both `common / release`(?) and `#master` Combinatorics.

``` jl
julia> using Combinatorics, Debugger

julia> @enter collect(combinations(1:3,2))
In collect(itr) at array.jl:601
 600  function collect(itr::Generator)
>601      isz = IteratorSize(itr.iter)
 602      et = @default_eltype(itr)
 603      if isa(isz, SizeUnknown)
 604          return grow_to!(Vector{et}(), itr)
 605      else

About to run: (getproperty)(<suppressed 212 bytes of output>, :iter)
1|debug> bp add Base.iterate(itr)
[ Info: added breakpoint for method iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})
1] iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})

1|debug> bp add Base.iterate(::Combinatorics.Combinations)
[ Info: added breakpoint for method iterate(::Combinatorics.Combinations)
1] iterate(::Base.Generator{Combinatorics.Combinations,getfield(Combinatorics, Symbol("#reorder#10")){UnitRange{Int64}}})
2] iterate(::Combinatorics.Combinations)

1|debug> c
3-element Array{Array{Int64,1},1}:
 [1, 2]
 [1, 3]
 [2, 3]
```

**Comment by KristofferC** [comment]
The error on master with Combinatorics is not Debugger's fault:

```
julia> Combinatorics.Combinations{UnitRange{Int64}}
ERROR: TypeError: in Type{...} expression, expected UnionAll, got Type{Combinatorics.Combinations}
Stacktrace:
 [1] top-level scope at REPL[3]:1
```

For the other problem... That is an embarrassing bug. Can you try with; https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/340?

**Comment by milesfrain** [comment]
Thanks. Everything is working well for me now.

**Pull request merged**